.github/workflows/tests.yaml

```diff
@@ -8,15 +8,26 @@ jobs:
     strategy:
       fail-fast: False
       matrix:
-        os: ['ubuntu-1804', 'ubuntu-2204', 'centos-7', 'centos-8']
+        os: ['ubuntu-1804-headless', 'ubuntu-2204-headless', 'centos-8']
```

Comment:
The VMs fail if the GUI is enabled so I've added headless versions for testing.
"centos-7" was failing with a weird LANG error but I don't this is worth investigating given it's age.